packages/venia-concept/src/components/CategoryList/categoryList.js

```diff
@@ -30,6 +30,21 @@ class CategoryList extends Component {
         ) : null;
     }
 
+    // map Magento 2.3.1 schema changes to v2.0.0 proptype shape to maintain backwards compatibility
+    mapCategory(categoryItem) {
+        categoryItem.productImagePreview.items = categoryItem.productImagePreview.items.map(
```

Comment:
I think we're pretty safe in assuming that none of `categoryItem`, `.productImagePreview`, or `.items` will be null or undefined, right?

Reply:
Correct, GraphQL will always return an array.

Reply:
Small but important thing: multiple components may use this object, so we shouldn't mutate it.

Immutable version:

```js
mapCategory(categoryItem) {
  const { items } = categoryItem.productImagePreview;
  return {
    ...categoryItem,
    productImagePreview: {
      items: items.map(item => {
        const { small_image } = item;
        return {
          ...item,
          small_image: typeof small_image === 'object' ? small_image.url : small_image
        }
      })
    }   
  }
}
```